### PR TITLE
Move splunk data from cold to frozen on s3 bucket

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by nadeemshahzad** [comment]
@edx/devops Please review

File: playbooks/roles/splunk-server/templates/opt/splunk/bin/coldToFrozenS3.j2 [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
```
Comment (fredsmith) [inline playbooks/roles/splunk-server/templates/opt/splunk/bin/coldToFrozenS3.j2]: should be:
```
#! /usr/bin/env bash
```
for portability

File: playbooks/roles/splunk-server/templates/opt/splunk/bin/coldToFrozenS3.j2 [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+s3_backup_bucket="{{ SPLUNK_FROZEN_DB_S3_BUCKET }}"
+
+if [ $# -lt 1 ]; then
+    echo "Usage: /bin/bash coldToFrozenS3 <bucket_dir_to_archive>"
```
Comment (fredsmith) [inline playbooks/roles/splunk-server/templates/opt/splunk/bin/coldToFrozenS3.j2]: no need to specify the interpreter.

**Review by fredsmith — APPROVED** [review]
two nits, otherwise LGTM

File: playbooks/roles/splunk-server/defaults/main.yml [hunk 3]
```diff
@@ -112,3 +112,6 @@ splunk_cert_path: 'etc/auth/edxcerts'
 splunk-server_debian_pkgs: []
 
 splunk-server_redhat_pkgs: []
+
+#S3 bucket to store data rolled from splunk cold bucket
+SPLUNK_FROZEN_DB_S3_BUCKET: "{{ SPLUNK_FROZEN_DB_DATA_S3_BUCKET }}"
```
Comment (fredsmith) [inline playbooks/roles/splunk-server/defaults/main.yml]: I think this should have a default value

**Review by fredsmith — CHANGES_REQUESTED** [review]
I think this will error without a default value.  Alternately, make installing your cold to frozen script conditional based on this variable being defined.

**Comment by nadeemshahzad** [comment]
@fredsmith Can you please review the changes?

**Review by fredsmith — APPROVED** [review]

**Pull request merged**